bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -768,6 +786,17 @@ private static boolean checkIsLikeMapType(Object sourceValue, BMapType targetTyp
         return true;
     }
 
+    private static boolean checkIsLikeTableType(Object sourceValue, BTableType targetType,
+                                              List<TypeValuePair> unresolvedValues) {
+        if (!(sourceValue instanceof TableValue)) {
+            return false;
+        }
+
+        BTableType tableType = (BTableType) ((TableValue) sourceValue).getType();
+
+        return tableType.getConstrainedType() == targetType.getConstrainedType();
+    }
+
     private static boolean checkIsLikeJSONType(Object sourceValue, BType sourceType, BJSONType targetType,
                                                List<TypeValuePair> unresolvedValues) {
         if (sourceType.getTag() == TypeTags.ARRAY_TAG) {
```

Comment:
Not related to this pr, theres an unused method `checkTableConstraints()` in the TypeChecker class. shall we remove it if no longer used?

Reply:
fixed